testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/OIDCAdvancedRequestParamsTest.java

```diff
@@ -631,6 +631,56 @@ public void requestObjectSupersedesQueryParameter() throws Exception {
         clientResource.update(clientRep);
     }
 
+    @Test
+    public void requestObjectClientIdAndResponseTypeTest() throws Exception {
+        oauth.stateParamHardcoded("some-state");
+
+        // Test that "client_id" mandatory in the query even if set in the "request" object
+        TestOIDCEndpointsApplicationResource oidcClientEndpointsResource = testingClient.testApp().oidcClientEndpoints();
+        oidcClientEndpointsResource.setOIDCRequest("test", "test-app", oauth.getRedirectUri(), "10", "some-state", "none");
+        oauth.request(oidcClientEndpointsResource.getOIDCRequest());
+        oauth.clientId(null);
+        oauth.openLoginForm();
+        errorPage.assertCurrent();
+
+        // Test that "response_type" mandatory in the query even if set in the "request" object
+        oidcClientEndpointsResource.setOIDCRequest("test", "test-app", oauth.getRedirectUri(), "10", "some-state", "none");
```

Comment:
This line and the following line might not be needed.

Reply:
Thanks for catching this. I've removed those lines. I've used separate commit, just so you can easily re-review that just this was changed. I plan to squash the commits before merging this.